### PR TITLE
EO-2891 Installer step reduction

### PR DESCRIPTION
**Pull request by hjaltisan**
We now have a new introduction page that replaces all pages before the perform installation step (except for the license page)
This new page shows the following:
- Where the Launcher will be installed
- Browse button to change that
- If redists will be installed
- How much space the installation will take

Additionally scripts are now loaded before showing the first page, and thus eliminating the meta progress bar that used to happen on next.

---
Tickets related to this are the following

EO-10460
- Moved script initialization to when entering the Introduction page instead of when pressing next

EO-10478
 - Removed ComponentSelection page
 - Removed StartMenuSelection page
 - Removed ReadyForInstallation page
 - scStartMenuDir now has the full path on startup

EO-4812
- Added a custom introduction page that replaces Introduction
- Added target directory to the custom intro page
- Added redist will be installed

EO-10818
- Added space required information

EO-11132
- Only show what should be shown on introduction page

EO-10820
- Added translations for the custom intro page

EO-11120
- Validate directory on change

EO-11121
- Fixed color of error text

EO11132
- First page layout cleanup

### CODE REVIEW

**Comment by hjaltisan** [comment]
There is a bit of commented out code. This is the original Qt FRW code for the original Introduction page. I left it in there so that we have it if we need to go back to it, and because the plan is (if the time ever comes) to create a cli argument for the framework, to tell it whether it should use the old introduction page or the new. And if that time never comes, then at least I might get the chance to change this into if else, with a hardcoded variable somewhere that is set to using the new introduction page

**Comment by hjaltisan** [comment]
Added a -xi|--custom-intro command line argument to binarycreator so that it knows which Introduction page to use. This way we still support the built in Introduction page

File: src/libs/installer/packagemanagercore_p.cpp [hunk 1]
```diff
@@ -2040,7 +2052,7 @@ void PackageManagerCorePrivate::registerMaintenanceTool()
     settings.setValue(QLatin1String("DisplayIcon"), maintenanceTool);
     settings.setValue(scPublisher, m_data.value(scPublisher));
     settings.setValue(QLatin1String("UrlInfoAbout"), m_data.value(QLatin1String("Url")));
-    settings.setValue(QLatin1String("Comments"), m_data.value(scTitle));
+    // settings.setValue(QLatin1String("Comments"), m_data.value(scTitle));
```
Comment (rosabj) [inline src/libs/installer/packagemanagercore_p.cpp]: Could you remove rather than comment out?

Comment (hjaltisan) [reply src/libs/installer/packagemanagercore_p.cpp]: Better?

**Comment by BaldvinTh** [comment]
No complaints here

**Review by rosabj — APPROVED** [review]

**Pull request merged**